src/core/abstracts/SablierLockupBase.sol

```diff
@@ -31,13 +31,13 @@ abstract contract SablierLockup is
                                   STATE VARIABLES
     //////////////////////////////////////////////////////////////////////////*/
 
-    /// @inheritdoc ISablierLockup
+    /// @inheritdoc ISablierLockupBase
     UD60x18 public constant override MAX_BROKER_FEE = UD60x18.wrap(0.1e18);
```

Comment:
These constants are specific to Lockup contracts. For example, we may have a lockup contract with a different max fee. Therefore, they should be a part of the this contract and not the libraries. Another benefit is that it makes libraries re-usable across future release of lockups.

Reply:
>These constants are specific to Lockup contracts

Max count, yes, since it can have different values accros chains, but why the max broker fee also?

> For example, we may have a lockup contract with a different max fee.

It is a `constant` and not `immutable`, which means a new contract must also be implemented declaring the `MAX_BROKER_FEE` with a new value. 

IMO, it is very unlikely to have different values for max broker fees; otherwise, we would have declared it as _immutable_.


The only purpose of this constant is to pass it in the Helpers.check() functions.


---

q: if this constant is declared in the library, would it be visible in Etherscan under the "Read Contract" category?

Reply:
>  it is very unlikely to have different values for max broker fees

I do not know why its not immutable but a constant. But my point is that Max Broker Fee and Max Protocol Fee are something that are specific to the protocol. On the other hand, libraries contain view only functions that take in some values, do some computations and return a new value.

Also, a user will be interested into getting max broker value from the lockup contract instead of going to the library address and checking the max value there. Also libraries are not allowed to have state variables and a constant is also a state variable (an immutable one). So imo it goes against the definition of a library.

> if this constant is declared in the library, would it be visible in Etherscan under the "Read Contract" category?

Should be.

Reply:
@andreivladbrg this is the last comment thats awaiting your reply.

Reply:
>I do not know why its not immutable but a constant. 

it is more a business decision. but IMO practically i don't see a scenario where we would want a fee > 10%, so we can keep it as constant

> But my point is that Max Broker Fee and Max Protocol Fee are something that are specific to the protocol.

yeah, fair point

> On the other hand, libraries contain view only functions that take in some values, do some computations and return a new value.

yes, they check against a value passed against a constant

also, having constants in libraries is a common thing: 

https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/libraries/TickMath.sol#L8-L16


> Also libraries are not allowed to have state variables and a constant is also a state variable (an immutable one). So imo it goes against the definition of a library.

constant and immutable are a part of the bytecode, no matter where that is placed (not in storage). so libraries are allowed to have "state variables". 

---

this comment: "Also, a user will be interested into getting max broker value from the lockup contract instead of going to the library address and checking the max value there"

and this one: "Should be."

i think they are in contradictory.

---

since it is more a business decision, and it's something specific to the protocol let's keep it here

Reply:
My arguments were bad. I was wrong to say that libraries can't have constant whereas they can.

I was against having broker fee constant in `VestingMath` library instead of Lockup contract because of the reason that a broker fee is more relevant to the lockup protocol than the library itself. So its about the placement of the variable. An example is `MAX_COUNT` which is placed in in `SablierLockup` and not in `SablierLockupBase` because its more appropriate for that contract since the segment length is compared against it.

> and this one: "Should be."

I thought your question was if Etherscan would show this value under "Read contract" category on the `VestingMath` library page for which I said "should be". Now I think you meant if it would show up on the Lockup contract page for which I would say "I don't know". Since the library would be deployed at a separate address, the Lockup contract might not show the constant value from the library.